src/Traits/StrictIterableTrait.php

```diff
@@ -253,4 +253,12 @@ public function concatAll()
 
         return $results;
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function reduce(callable $callback, $initial = null)
+    {
+        return array_reduce($this->toArray(), $callback, $initial);
```

Comment:
So this would only work for array data structures.... If you could write manually would be better